core/src/main/java/org/apache/accumulo/core/metadata/schema/Ample.java

```diff
@@ -119,6 +119,15 @@ public static DataLevel of(TableId tableId) {
    */
   TabletMetadata readTablet(KeyExtent extent, ColumnType... colsToFetch);
 
+  /**
+   * Entry point for reading multiple tablets' metadata. Generates a TabletsMetadata builder object
+   * and assigns the AmpleImpl client to that builder object. This allows readTablets() to be called
+   * from a ClientContext. Associated methods of the TabletsMetadata Builder class are used to
+   * generate the metadata.
+   */
+
+  AmpleImpl.Builder readTablets();
```

Comment:
This exposes the Impl class on the interface.  This breaks the contract of the interface.

Reply:
I think this method should return `TabletsMetadata.TableOptions`.  That will limit the next method calls to what is needed for the fluent entry point and help prevent coding errors.

We want to select the table first like this:
```
ample.readTablets().forTable(tableId).fetch(LOCATION, PREV_ROW).build();
```
And not be able to skip the required parameters like this:
```
ample.readTablets().build(); //coding error
```

Reply:
Incorporated suggestions on latest push.